Assets/MixedRealityToolkit/Utilities/BuildAndDeploy/UwpAppxBuildTools.cs

```diff
@@ -359,15 +389,53 @@ private static string GetMSBuildLoggingCommand(string logDirectory, string logFi
         }
 
         /// <summary>
-        /// Adds the 'Gaze Input' capability to the manifest.
+        /// An overload of AddResearchModeCapability that will read the AppX manifest
+        /// from the build output and update the manifest file with the rescap aka
+        /// research mode capability.
         /// </summary>
-        /// <remarks>
-        /// This is a workaround for versions of Unity which don't have native support
-        /// for the 'Gaze Input' capability in its Player Settings preference location.
-        /// Note that this function is only public to poke a hole for testing - do not
-        /// take a dependency on this function.
-        /// </remarks>
-        public static void AddGazeInputCapability(XElement rootNode)
+        /// <param name="buildInfo">An IBuildInfo containing a valid OutputDirectory</param>
+        public static void AddCapabilities(IBuildInfo buildInfo)
+        {
+            string manifestFilePath = GetManifestFilePath(buildInfo);
+            if (manifestFilePath == null)
+            {
+                throw new FileNotFoundException("Unable to find manifest file");
+            }
+
+            var rootElement = XElement.Load(manifestFilePath);
+            var uwpBuildInfo = buildInfo as UwpBuildInfo;
+
+            Debug.Assert(uwpBuildInfo != null);
+            if (uwpBuildInfo.GazeInputCapabilityEnabled)
+            {
+                AddGazeInputCapability(rootElement);
+            }
+
+            if (EditorUserBuildSettings.wsaSubtarget == WSASubtarget.HoloLens)
+            {
+                if (uwpBuildInfo.ResearchModeCapabilityEnabled)
+                {
+                    AddResearchModeCapability(rootElement);
+                }
+
+                if (EditorUserBuildSettings.wsaGenerateReferenceProjects && uwpBuildInfo.AllowUnsafeCode)
+                {
+                    AllowUnsafeCode(rootElement);
```

Comment:
I think that rootElement in this location is the AppX manifest, not the Assembly-CSharp manifest (which is the type of manifest that unsafe code has to be on) - I think that this may have gotten a lost in the refactor, and I can definitely take a little bit of blame here for not being a little clearer in my request!

Reply:
i.e. to be clear, I think that AllowUnsafeCode should probably be like what it was before, and be called as a same-level sort of sibling do AddCapabilities (where AddCapabilites is called).

```diff
-                    if (uwpBuildInfo.GazeInputCapabilityEnabled)
-                    {
-                        UwpAppxBuildTools.AddGazeInputCapability(uwpBuildInfo);
-                    }
+                    UwpAppxBuildTools.AddCapabilities(uwpBuildInfo);
+                    UwpAppxBuildTools.AllowUnsafeCode(uwpBuildInfo);

```

Reply:
yeah...that is some `refactor copy&paste past midnight`-error 😅 sorry about that.

Reply:
I added a method like `UpdateAppxManifest` that can be used as a "central" point for changes that need to be made in the project file.